### PR TITLE
PRMT-4669 Repository E2E testing fixes

### PR DESCRIPTION
**Pull request by AndyFlintNHS**

### CODE REVIEW

File: src/test/resources/templates/ehr-request.hbs [hunk 1]
```diff
@@ -1 +1 @@
...
...oot=\"1.2.826.0.1285.0.2.0.107\" extension=\"200000000631\" /></agentSystemSDS></AgentSystemSDS></author1><subject type=\"ActRelationship\" typeCode=\"SUBJ\" contextConductionInd=\"false\"><EhrRequest type=\"ActHeir\" classCode=\"EXTRACT\" moodCode=\"RQO\"><id root=\"FFFB3C70-0BCC-4D9E-A441-7E9C41A897AA\" /><recordTarget type=\"Participation\" typeCode=\"RCT\"><patient type=\"Patient\" classCode=\"PAT\"><id root=\"2.16.840.1.113883.2.1.4.1\" extension=\"{{nhsNumber}}\" /></patient></recordTarget><author type=\"Participation\" typeCode=\"AUT\"><AgentOrgSDS type=\"RoleHeir\" classCode=\"AGNT\"><agentOrganizationSDS type=\"Organization\" classCode=\"ORG\" determinerCode=\"INSTANCE\"><id root=\"1.2.826.0.1285.0.1.10\" extension=\"{{senderOdsCode}}\" /></agentOrganizationSDS></AgentOrgSDS></author><destination type=\"Participation\" typeCode=\"DST\"><AgentOrgSDS type=\"RoleHeir\" classCode=\"AGNT\"><agentOrganizationSDS type=\"Organization\" classCode=\"ORG\" determinerCode=\"INSTANCE\"><id root=\"1.2.826.0.1285.0.1.10\" extension=\"{{receivingOdsCode}}\" /></agentOrganizationSDS></AgentOrgSDS></destination></EhrRequest></subject></ControlActEvent></RCMR_IN010000UK05>","attachments":[]}
```
Comment (martin-nhs) [inline src/test/resources/templates/ehr-request.hbs:1]: Just for reference, `extension=\"{{sendingOdsCode}}\"` has now been changed to `extension=\"{{senderOdsCode}}\"`.

File: src/test/java/uk/nhs/prm/e2etests/test/ArchivedE2ETest.java [hunk 2]
```diff
@@ -138,34 +139,36 @@ void shouldReceivingAndTrackAllLargeEhrFragments_DevAndTest() {
...
-        Acknowledgement acknowledgement = await().atMost(30, TimeUnit.SECONDS)
-                .with().pollInterval(2, TimeUnit.SECONDS)
-                .until(() -> ehrOutService.findAcknowledgementByMessageId(ackMessageId), Objects::nonNull) ;
-
-        // then
-        String actualTypeCode = acknowledgement.getAcknowledgementTypeCode();
-        assertThat(actualTypeCode).isEqualTo(expectedTypeCode);
-        log.info("The acknowledgement typeCode of {} is {}.", ackMessageId, actualTypeCode);
-    }
+    // TODO PRMT-4669 commented out due to UUID/STRING discrepancy - archived test, not worth fixing at the moment
```
Comment (martin-nhs) [inline src/test/java/uk/nhs/prm/e2etests/test/ArchivedE2ETest.java:142]: The ticket on this `TODO` is referring to this PR (PRMT-4669), therefore should be changed to another ticket to address this.

File: src/main/java/uk/nhs/prm/e2etests/queue/SimpleAmqpQueue.java [hunk 3]
```diff
@@ -95,7 +95,7 @@ private Producer createProducer() {
             }
         }
 
-        throw new GenericException(this.getClass().getName(), "Failed to connect to any endpoint");
+        throw new GenericException(this.getClass().getName(), "Failed to connect to any endpoint, this can occur when you are not connected to the VPN. If you are connected to the VPN, try restart the AMQP broker.");
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/e2etests/queue/SimpleAmqpQueue.java:98]: Possibly scope creep, though if we can move away from `GenericException` to something more explicit - such as `AmqpException`. The reflection seems overkill too.

File: src/main/java/uk/nhs/prm/e2etests/property/TestConstants.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        log.info("largeEhrCoreMessageId: " + largeEhrCoreMessageId);
+        log.info("fragment1MessageId: " + fragment1MessageId);
+        log.info("fragment2MessageId: " + fragment2MessageId);
+        log.info("nemsMessageId: " + nemsMessageId);
+        log.info("senderOdsCode: " + senderOdsCode);
+        log.info("recipientOdsCode: " + recipientOdsCode);
+        log.info("asidCode: " + asidCode);
+        log.info("==========TEST CONSTANTS==========");
+        log.info("ANY VARIABLES LISTED BELOW HAVE BEEN DEFINED WITHIN THE TEST");
+        log.info("IF A VARIABLE FROM ABOVE (SUCH AS senderOdsCode, FOR EXAMPLE) IS REPEATED BELOW, BELOW SHOULD BE CONSIDERED THE SOURCE OF TRUTH");
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/e2etests/property/TestConstants.java:53]: Override a toString() and log that?

File: src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java [hunk 5]
```diff
@@ -4,19 +4,24 @@
...
-    /**
-     * @deprecated PRMT-4670 old Transfer Tracker DB logic - to be removed
-     */
-    @Deprecated(since="2.0.0", forRemoval = true)
+    @Deprecated(since="2.0.0")
     @Value("${aws.configuration.databaseNames.oldTransferTrackerDb}")
     private String oldTransferTrackerDbName;
 
+    // TODO - PRMT-4663 - USE @Value WHERE IT IS REQUIRED,
+    // TODO - PRMT-4663 - LET'S NOT COUPLE THESE NAMES!
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java:19]: Should these be addressed with this ticket, PRMT-4663 is the parent!

File: src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java [hunk 6]
```diff
@@ -4,19 +4,24 @@
...
     @Value("${aws.configuration.databaseNames.transferTrackerDb}")
     private String transferTrackerDbName;
 
+    // TODO - PRMT-4663 - USE @Value WHERE IT IS REQUIRED,
+    // TODO - PRMT-4663 - LET'S NOT COUPLE THESE NAMES!
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java:24]: Should these be addressed with this ticket, PRMT-4663 is the parent!

File: src/main/java/uk/nhs/prm/e2etests/queue/SimpleAmqpQueue.java (hunk 3 above)
Comment (AndyFlintNHS) [reply src/main/java/uk/nhs/prm/e2etests/queue/SimpleAmqpQueue.java:98]: I completely agree with your points. It's definitely scope creep in this context, there's a tech debt item to address all uses of 'GenericException'.

File: src/main/java/uk/nhs/prm/e2etests/property/TestConstants.java (hunk 4 above)
Comment (AndyFlintNHS) [reply src/main/java/uk/nhs/prm/e2etests/property/TestConstants.java:53]: Unable to Override toString() statically and unable to reference a non-static toString from a static context. 

I did look at breaking it into a single log statement with a multi-line string, but injecting the values into it this way without using the new Java 21 STR string template (E2E tests still on Java 17) honestly just makes it less readable to me.

File: src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java (hunk 5 above)
Comment (AndyFlintNHS) [reply src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java:19]: I've removed these TODO messages, they're on a deprecated Component that is only used in logic that is to be removed, and the active suspensions (which is also going to be removed eventually).

File: src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java (hunk 6 above)
Comment (AndyFlintNHS) [reply src/main/java/uk/nhs/prm/e2etests/property/DynamoDbProperties.java:24]: I've removed these TODO messages, they're on a deprecated Component that is only used in logic that is to be removed, and the active suspensions (which is also going to be removed eventually).

**Review by martin-nhs — APPROVED** [review]
